src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -1058,6 +1058,42 @@ public void ConnectionsPooledThenDisposed_NoUnobservedTaskExceptions(bool secure
                 return SuccessExitCode;
             }, secure.ToString()).Dispose();
         }
+
+        [Fact]
+        public async Task ProxyAuth_SameConnection_Succeeds()
+        {
+            Task serverTask = LoopbackServer.CreateServerAsync(async (proxyServer, proxyUrl) =>
+            {
+                string responseBody =
+                        "HTTP/1.1 407 Proxy Auth Required\r\n" +
+                        $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
+                        "Proxy-Authenticate: Basic\r\n" +
+                        "Content-Length: 0\r\n" +
+                        "\r\n";
+
+                using  (HttpClientHandler handler = new HttpClientHandler())
+                {
+                    handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, new NetworkCredential("abc", "def"));
+
+                    using (var client = new HttpClient(handler))
+                    {
+                        Task<string> request = client.GetStringAsync($"http://notarealserver.com/");
+
+                        await proxyServer.AcceptConnectionAsync(async connection =>
+                        {
+                            // Get first request, no body for GET.
+                            await connection.ReadRequestHeaderAndSendCustomResponseAsync(responseBody).ConfigureAwait(false);
+                            // Client should send another request after being rejected with 407.
+                            await connection.ReadRequestHeaderAndSendResponseAsync(content:"OK").ConfigureAwait(false);
+                        });
+
+                        String response = await request;
+                        Assert.Equal("OK", response);
+                    }
+                }
+            });
+            await TaskTimeoutExtensions.WhenAllOrAnyFailed(new Task[] { serverTask }, TestHelper.PassingTestTimeoutMilliseconds);
```

Comment:
Since there is only one task here, you should use .TimeOutAfter.

See: https://github.com/dotnet/corefx/blob/de2c36b14b827379fb4fd0484c712e7dfccfec60/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L3247